experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -297,8 +297,25 @@ export interface MapNode<in out TSchema extends MapSchema> extends TreeNode {
 		thisArg?: any,
 	): void;
 
-	// TODO: Add `set` method when FieldKind provides a setter (and derive the type from it).
-	// set(key: string, content: FlexibleFieldContent<TSchema["mapFields"]>): void;
+	/**
+	 * Sets a new value associated with `key`.
+	 * @param key - The key under which the new value will be set.
+	 * @param content - The new value to associate with `key`.
+	 */
+	set(key: string, content: FlexibleFieldContent<TSchema["mapFields"]>): void; // TODO: useful to return itself like JS map?
+
+	/**
+	 * Deletes the value associated with `key` from the map, if it exists.
+	 *
+	 * @remarks
+	 * Note: unlike JavaScript's Map API, this method does not return a flag indicating whether or not the value was
+	 * deleted. This is an intentional design choice. Since this data structure is distributed in nature, it isn't
+	 * possible to tell whether or not the item was deleted as a result of this method call. Returning a "best guess"
+	 * is more likely to create issues / promote bad usage patterns than offer useful information.
```

Comment:
imo nit: I would omit the closing sentence of this doc. I think it's useful to tell users the mechanical reasons why we can't support an API (the sentence prior) but going into the psychology of it is more the domain of tutorials or blog posts than doc comments.

I would also remove the sentence that says "This is an intentional design choice". It will be apparent that it is an intentional choice, because we wrote this doc comment about it :D

Reply:
Yeah, that's fair. I still think the context is useful for members of our team, though. I moved the latter sentence to a `privateRemarks` block. Does that seem reasonable to you?

Reply:
Yea that's great, good idea.